AllJoyn/Platform/BridgeHostStartupTask/Program.cs

```diff
@@ -0,0 +1,140 @@
+                            Console.ForegroundColor = ConsoleColor.Green;
+                            connection.RequestReceived += Connection_RequestReceived;
+                            connection.ServiceClosed += Connection_ServiceClosed;
+                            Console.WriteLine("Connection established - waiting for requests");
+                            Console.WriteLine();
+                            break;
+                        case AppServiceConnectionStatus.AppNotInstalled:
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine("The app AppServicesProvider is not installed.");
+                            break;
+                        case AppServiceConnectionStatus.AppUnavailable:
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine("The app AppServicesProvider is not available.");
+                            break;
+                        case AppServiceConnectionStatus.AppServiceUnavailable:
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine(string.Format("The app AppServicesProvider is installed but it does not provide the app service {0}.", connection.AppServiceName));
+                            break;
+                        case AppServiceConnectionStatus.Unknown:
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine(string.Format("An unkown error occurred while we were trying to open an AppServiceConnection."));
+                            break;
+                    }
+                }
+
+                Thread.Sleep(5000); // Do Health checks every 5 seconds
+            } // while()
+        }
+
+        private static void Connection_ServiceClosed(AppServiceConnection sender, AppServiceClosedEventArgs args)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            isConnectionClosed = true;
+            connection = null;
+            Console.WriteLine(string.Format("AppServiceConnection is closed!: Status : " + args.Status));
+        }
+
+        /// <summary>
+        /// Receives message from UWP app and sends a response back
+        /// </summary>
+        private static void Connection_RequestReceived(AppServiceConnection sender, AppServiceRequestReceivedEventArgs args)
+        {
+            string key = args.Request.Message.First().Key;
+            string value = args.Request.Message.First().Value.ToString();
+            Console.ForegroundColor = ConsoleColor.Cyan;
+            Console.WriteLine(string.Format("Received message '{0}' with value '{1}'", key, value));
+            //if (key == "request")
+            {
+                ValueSet valueSet = new ValueSet();
+                valueSet.Add("response", value.ToUpper());
+                Console.ForegroundColor = ConsoleColor.White;
```

Comment:
#Resolved

---
In reply to: [75205216](https://github.com/tjaffri/msiot-samples/pull/9#discussion_r75205216) [](ancestors = 75205216,75187717)